src/kaboom.ts

```diff
@@ -6394,12 +6475,44 @@ export default (gopt: KaboomOpt = {}): KaboomCtx => {
 		app.keyState.down.forEach((k) => game.ev.trigger("keyDown", k))
 		app.mouseState.down.forEach((k) => game.ev.trigger("mouseDown", k))
 		app.virtualButtonState.down.forEach((k) => game.ev.trigger("virtualButtonDown", k))
+		
 		for (const gamepad of navigator.getGamepads()) {
-			// TODO
+			if(!gamepad) return // the gamepad can return null if isn't a gamepad or is disconnected
+			let map = GAMEPAD_MAPS[gamepad.id]
+			if(!map) map = GAMEPAD_MAPS["default"]
+
+			for(let i = 0; i < gamepad.buttons.length; i++) {
+				if(gamepad.buttons[i].pressed) {
+					if(!app.gamepadButtonState.down.has(map.buttons[i])) {
+						app.gamepadButtonState.press(map.buttons[i])
+						game.ev.trigger("gamepadButtonPress", map.buttons[i])
+					}
+
+					game.ev.trigger("gamepadButtonDown", map.buttons[i])
+				}
+				else {
+					if(app.gamepadButtonState.pressed.has(map.buttons[i]) || app.gamepadButtonState.down.has(map.buttons[i])) {
+						app.gamepadButtonState.release(map.buttons[i])
+						game.ev.trigger("gamepadButtonRelease", map.buttons[i])
+					}
+				}
+			}
+
+			for(const axeName in map.axes) {
```

Comment:
Nit: the singular of axes is axis i think

Reply:
Ty!